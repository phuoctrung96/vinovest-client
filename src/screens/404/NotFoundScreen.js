import React from "react";
import { I18nLink } from "#localization/localizedRouter";
import { ReactComponent as NotFoundIcon } from "./assets/404_bottle.svg";
import StyledNotFoundScreen from "./NotFoundScreen.styled";
import { createGlobalStyle } from "styled-components";
import { useTranslation } from "react-i18next";

const GlobalStyle = createGlobalStyle` 
footer{
display: none !important;
}`;

const NotFoundScreen = () => {
    const { t } = useTranslation(["404"]);
    return (
        <StyledNotFoundScreen>
            <GlobalStyle />
            <NotFoundIcon />
            <h1 data-testid="title">{t("title")}</h1>
            <h6 data-testid="subtitle">{t("subtitle")}</h6>
            <I18nLink to="/" data-testid="back">
                {t("back_link")}
            </I18nLink>
        </StyledNotFoundScreen>
    );
};

export default NotFoundScreen;
