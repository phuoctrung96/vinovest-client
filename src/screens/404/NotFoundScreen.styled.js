import styled from "styled-components";

const StyledNotFoundScreen = styled.div`
    height: calc(100vh - 64px);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    h1 {
        font-weight: 500;
        font-size: 36px;
        line-height: 30px;
        margin-bottom: 25px;
        margin-top: 35px;
    }
    h6 {
        font-weight: normal;
        font-size: 18px;
        line-height: 26px;
        margin-top: 0;
        margin-bottom: 45px;
    }

    a {
        width: 202px;
        border: 1px solid #000000;
        box-sizing: border-box;
        color: #000000;
        text-decoration: none;
        text-align: center;
        line-height: 50px;
        font-weight: 500;
        font-size: 14px;
        letter-spacing: 0.05em;
        text-transform: uppercase;

        :hover {
            cursor: pointer;
            box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.3);
        }
    }
`;

export default StyledNotFoundScreen;
