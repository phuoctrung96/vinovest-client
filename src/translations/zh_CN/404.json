{
    "back_link": "转到首页",
    "subtitle": "该页面缺失或页面链接有误。",
    "title": "找不到该页面"
}