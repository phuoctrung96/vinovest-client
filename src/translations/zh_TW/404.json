{
    "back_link": "返回首頁",
    "subtitle": "頁面丟失或您錯誤地組裝了鏈接",
    "title": "找不到網頁"
}