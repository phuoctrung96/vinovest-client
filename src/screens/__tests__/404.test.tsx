import React from "react";
import { render } from "@testing-library/react";
import NotFoundScreen from "../404/NotFoundScreen";
import { withContexts } from "#shared/testing/wrappers";
import { title as t_title, subtitle as t_subtitle, back_link } from "#translations/en/404.json";

describe("NotFoundScreen component", () => {
    let component;
    beforeAll(() => {
        const { component: Comp } = withContexts(NotFoundScreen);
        component = Comp;
    });
    it("should properly render the default value state", () => {
        const { baseElement, getByTestId } = render(component());
        const title = getByTestId("title");
        const subtitle = getByTestId("subtitle");
        const link = getByTestId("back");

        expect(baseElement).toMatchSnapshot();
        expect(title.textContent).toEqual(t_title);
        expect(subtitle.textContent).toEqual(t_subtitle);
        expect(link.textContent).toEqual(back_link);
    });
});
