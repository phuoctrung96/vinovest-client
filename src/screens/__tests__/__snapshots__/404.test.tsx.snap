// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundScreen component should properly render the default value state 1`] = `
.c0 {
  height: calc(100vh - 64px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 h1 {
  font-weight: 500;
  font-size: 36px;
  line-height: 30px;
  margin-bottom: 25px;
  margin-top: 35px;
}

.c0 h6 {
  font-weight: normal;
  font-size: 18px;
  line-height: 26px;
  margin-top: 0;
  margin-bottom: 45px;
}

.c0 a {
  width: 202px;
  border: 1px solid #000000;
  box-sizing: border-box;
  color: #000000;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: 50px;
  font-weight: 500;
  font-size: 14px;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  text-transform: uppercase;
}

.c0 a:hover {
  cursor: pointer;
  box-shadow: 0px 4px 20px rgba(0,0,0,0.3);
}

<body>
  <div>
    <div
      class="c0"
    >
      <div />
      <h1
        data-testid="title"
      >
        Page Not Found
      </h1>
      <h6
        data-testid="subtitle"
      >
        The page is missing or you assembled the link incorrectly
      </h6>
      <a
        data-testid="back"
        href="/"
      >
        Back to Home
      </a>
    </div>
  </div>
</body>
`;
